### PR TITLE
Adds default container annotation to Prometheus pod

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Adds [default container annotation](https://kubernetes.io/docs/reference/labels-annotations-taints/#kubectl-kubernetes-io-default-container) to Prometheus pod, so that kubectl fetches logs from the Prometheus container if no container is specified. Requires kubectl >= 1.21.0.

Without patch:

```
$ kubectl logs -f prometheus-prometheus-0
error: a container name must be specified for pod prometheus-prometheus-0, choose one of: [prometheus config-reloader]
```

With patch:

```
$ kubectl logs -f prometheus-prometheus-0
level=info ts=2021-04-09T19:36:12.550Z caller=main.go:418 msg="Starting Prometheus" version="(version=2.26.0, branch=HEAD, revision=3cafc58827d1ebd1a67749f88be4218f0bab3d8d)"
...
```
```
$ kubectl logs -f prometheus-prometheus-0 prometheus
level=info ts=2021-04-09T19:36:12.550Z caller=main.go:418 msg="Starting Prometheus" version="(version=2.26.0, branch=HEAD, revision=3cafc58827d1ebd1a67749f88be4218f0bab3d8d)"
```
```
$ kubectl logs -f prometheus-prometheus-0 config-reloader
level=info ts=2021-04-09T19:36:11.481259701Z caller=main.go:147 msg="Starting prometheus-config-reloader" version="(version=0.46.0, branch=refs/tags/pkg/client/v0.46.0, revision=7f94a06b86d41c20176f0d5b53aa0100fdc361e6)"
...
```

tl;dr - i always forget to specify the prometheus container and this makes that so much more awesome

**Release Note Template (will be copied)**

```release-note:enhancement
Add default container annotation to Prometheus pod
```

### CODE REVIEW

**Review by paulfantom — APPROVED** [review]
TIL that this is possible :exploding_head: 

Awesome! :shipit:

**Comment by paulfantom** [comment]
We should add this for Alertmanger too.

**Comment by JosephSalisbury** [comment]
Yeah, I have a follow up PR incoming :D

**Comment by JosephSalisbury** [comment]
See https://github.com/prometheus-operator/prometheus-operator/pull/3978

**Review by QuentinBisson — APPROVED** [review]

**Review by simonpasquier — APPROVED** [review]
